Python/marshal.c

```diff
@@ -310,7 +310,9 @@ w_PyLong(const PyLongObject *ob, char flag, WFILE *p)
     }
     if (!long_export.digits) {
         int8_t sign = long_export.value < 0 ? -1 : 1;
-        uint64_t abs_value = Py_ABS(long_export.value);
+        uint64_t abs_value = long_export.value < -INT64_MAX
+            ? (uint64_t)INT64_MAX + (uint64_t)-(long_export.value + INT64_MAX)
+            : (uint64_t)Py_ABS(long_export.value);
```

Comment:
Can't we assume two's complement representation of integers and use simpler workaround?  As this fix span several places, I would prefer if it could be refactored to a separate macro.

I.e. something like:
```c
#define My_ABS(x, MAX) \
    ((x) < 0 ? ((x) >= -MAX ? -(x) : (U##MAX >> 1) + 1) : (x))
```

Nowadays two's complement is only case permitted by the C23 and is a de-facto standard.  Do you have some system in mind on which we should care?  I'm pretty sure all Tier 1-3 platforms fit to this picture.

Reply:
> All Tier 1-3 platforms perhaps fine with the current code.

(Yes, and I suspect tests might be redundant in fact.)

> Although for your proposition it generates more complex code.

A different version:
```c
#define __GMP_CAST(type, expr) ((type) (expr))
#define NEG_CAST(T,x) (- (__GMP_CAST (T, (x) + 1) - 1))
#define ABS_CAST(T,x) ((x) >= 0 ? __GMP_CAST (T, x) : NEG_CAST (T, x))
```
I found same approach in the GNU GMP, so just copied NIH code here.

> Anyway, the performance of this code is not critical (if there is any difference).

Your solution looks ok for me.  But in any case we should factor it to some macro.